### DISCUSSION TITLE
Version 3.5 released

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#3782] [Announcements]
Full release notes: [www.sumatrapdfreader.org/docs/Version-history](https://www.sumatrapdfreader.org/docs/Version-history)
To download: [www.sumatrapdfreader.org/download-free-pdf-viewer](https://www.sumatrapdfreader.org/download-free-pdf-viewer)

The big changes are:
* dark mode
* Arm 64-bit builds
* improvements / bug fixes to annotation editing
* support for `.avif` files

Detailed notes:
- Arm 64-bit builds
- dark mode (menu `Settings / Theme` or `Ctrl + K` command `Select next theme`)
  you can use `i` (invert colors) to match the background / text color of rendered
  PDF document. Due to technical limitations, it doesn't work well with images
- `i` (invert colors) is remembered in settings
- `CmdEditAnnotation` select annotation under cursor and open annotation edit window
- rename `CmdShowCursorPosition` => `CmdToggleCursorPosition`
- add `Annotations [ FreeTextColor, FreeTextSize, FreeTextBorderWidth ]` settings
- ability to move annotations. `Ctrl + click` to select annotation and them move via drag & drop
- add `CmdCommandPaletteOnlyTabs` command with `Alt + K` shortcut
- exit full screen / presentation modes via double click with left mouse button
- ability to drag out a tab to open it in new window
- support opening `.avif` images (including inside .cbz/,cbr files)
- respect image orientation `exif` metadata in .jpeg and .png images
- support Adobe Reader syntax for opening files `/A "page=<pageno>#nameddest=<dest>search=<string>`
- add `Next Tab` / `Prev Tab` commands with `Ctrl + PageUp` / `Ctrl + PageDown` shortcuts
- keep Home tab open; add `NoHomeTab` advanced option to d`isable that
- add context menu to tabs
- bugfix: handle files we can't open in `next file in folder` / `prev file in folder` commands
- command palette: when search starts with `>`, only show commands, not files (like in Visual Studio Code)
- add `reopen last closed` command (`Ctrl + Shift + T`, like in web browsers)
- add `clear history` command
- can send commands via [DDE](https://www.sumatrapdfreader.org/docs/DDE-Commands)
- added `CmdOpenWithExplorer`, `CmdOpenWithDirectoryOpus`, `CmdOpenWithTotalCommander`, `CmdOpenWithDoubleCommander` commands
- enable `CmdCloseOtherTabs`, `CmdCloseTabsToTheRight` commands from command palette
- recognize `PgUp` / `PgDown` and a few more in keyboard shortcuts
- add `-disable-auto-rotation` cmd-line print option

<hr /><em>This discussion was created from the release <a href='https://github.com/sumatrapdfreader/sumatrapdf/releases/tag/3.5rel'>3.5 release</a>.</em>

**Comment by Netkuszkusz** [comment]
The installer still doesn't work if I rename it. This is nonsense.

**Reply by GitHubRulesOK** [reply]
you did not say if your read the docs or the help command ? I have yet to install so let me test with a renamed copy.

**Reply by Netkuszkusz** [reply]
I list softwares on a specific application collection website with standardized names, that's why it sucks. I have 280 applications there and only this cannot be renamed.

**Reply by GitHubRulesOK** [reply]
works for me if iI run SPDF.exe
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/558d0e0b-d87b-4710-aa17-cbe2dc8491b3)

**Reply by Netkuszkusz** [reply]
![scr](https://github.com/sumatrapdfreader/sumatrapdf/assets/15225482/f7561e2c-2fa7-4a80-9ee3-ffe1f24fb883)

As I name it back to SumatraPDF-3.5-64-install.exe it works again.
The fix seems to be easy: don't try to do anything if I don't give it any argument, just run it as usual.

**Reply by GitHubRulesOK** [reply]
the installer needs the -install switch to know it is in installer mode (simply done by include switch in name) and  without it there is no support modules or valid command (it cannot in that case be run the same as the portable version) so errors

other apps have different ways of using name for installation control so for example https://miktex.org/howto/portable-edition

So if its to be run as an installer any rename with -install as part of the name does not require adding that switch
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/53e37f86-71e4-48bf-b053-a6367974320f)

**Reply by Netkuszkusz** [reply]
> installer needs the -install switch to know it is in installer mode

That's a dumb thing. The basic thing what an installer should do with any filename: fucking install, not crash.

**Comment by Ariesun24** [comment]
Uninstaller doesn't work

**Reply by kjk** [reply]
thanks for letting me know, should be fixed in 3.5.1

**Comment by jls28** [comment]
I found the home tab is useless and have a too much large width : 
a permanent button could replace it.

**Reply by kjk** [reply]
there's an `NoHomeTab` option to disable it:
```
keep Home tab open; add NoHomeTab advanced option to disable that
```

**Comment by CramericaIndustries** [comment]
Installed 3.5 (Windows) yesterday. Zooming into an image/graphic is now super slow/jerky.

**Reply by kjk** [reply]
please open a bug and provide step-by-step instructions to reproduce the issue, which version you're comparing to etc., as per https://www.sumatrapdfreader.org/docs/How-to-submit-bug-reports

**Comment by iffling** [comment]
Updated to newest version (3.5.2 64-bit, Win10 Pro for Workstations, 22H2, build 19045.3693, Feature Experience Pack 1000.19053.1000.0). Don't know what the old version was (3.4.something).

* It lost all settings from existing install, including any memory of what files I had open
* It will not retain new settings (e.g. location on screen, document history, document page, turning off tabs)
* It does not remember per-document settings, either, despite that being checked by default (probably because it isn't saving any settings - each time I launched it, it came up as if it were the first time)

I finally uninstalled the new version and installed 3.3.3 (using the last installer I still had on disk). Reverting to the older version remembered my settings and pages last opened, as well as screen position, and tabs off. It had an outdated listing of last-opened files, however.

If it matters, it is NOT being installed to the default folder, it is instead being installed (for all users) to C:\Utils\SumatraPDF, just as I have been doing since version 1.x.

**Reply by GitHubRulesOK** [reply]
When installing a copy as admin for "all users" it should be to the system all users "Program Files" location. and each user settings will by default be in their users %localappdata% directory (and SumatraPDF subfolder) so ensure normal users are not trying to use the administrators settings, they will by default each have their own.

**Reply by iffling** [reply]
That is irrelevant. It should not matter. You can specify the install directory right there in the options for installation, regardless of user or global, and it was supposed to upgrade the existing (global) install. It has worked fine this way for years and years, across scores of machines in dozens of different system environments, and in all major variants of Windows since SumatraPDF came out (during XP as I recall, based on what stuff I was working on then).

Reinstalling the older version everything works fine. It was only the newest version that lost everything and was unable to save new settings. (I have made it a practice to avoid the Program Files folders as much as possible ever since they were introduced. And my personal settings are in a folder outside the Users directory tree [e.g. C:\myusername, which nowadays is typically a mounted disk]. Again, has not been a problem, pretty much ever. Permissions are set on the folders, and registry and environment vars are set appropriately. I come from the *nix world and don't like the glomming of folders introduced with Win95.)

**Comment by Erwan62** [comment]
Hello
Is it normal that in annotation editing, when you click on delete (to edit your text), your annotation is entirely deleted?

Edit : After checking, the problem does not exist in 3.6 pre-release.

**Reply by kjk** [reply]
Yes, it was a bug in 3.5 and fixed in pre-release
